consolidate with FreeSurfer's LUT

m-reuter opened the pull request:
Adding labels from FreeSurfer (dev)

m-reuter committed:
consolidate with FreeSurfer's LUT

dkuegler commented:
Then our Hypothalamus numbers (#459) conflict with the FreeSurfer changes....


```
901  R-N.opticus                            70  130 180 0
902  L-N.opticus                            130 180 70  0
903  R-C.mammilare                          205 62  78  0
904  R-Optic-tract                          80  120 134 0
905  L-Optic-tract                          196 58  250 0
906  L-C.mammilare                          0   148 0   0
907  R-Chiasma-Opticum                      220 248 164 0
908  L-Chiasma-Opticum                      230 148 34  0
909  Ant-Commisure                          10  180 225 0
910  3rd-Ventricle                          118 0   100 0
911  R-Fornix                               122 200 120 0
912  L-Fornix                               236 13  176 0
914  Epiphysis                              204 182 142 0
916  Hypophysis                             119 159 176 0
917  Infundibulum                           220 216 20  0
922  Tuberal-Region                         120 60  110 0
926  L-Med-Hypothalamus                     165 255 0   0
927  L-Lat-Hypothalamus                     0   255 127 0
928  L-Ant-Hypothalamus                     165 42  42  0
929  L-Post-Hypothalamus                    255 215 0   0
936  R-Med-Hypothalamus                     115 255 0   0
937  R-Lat-Hypothalamus                     60  255 127 0
938  R-Ant-Hypothalamus                     165 142 42  0
939  R-Post-Hypothalamus                    255 170 20  0

# Below is the color table for olfactory bulb structures.
# See : Automated Olfactory Bulb Segmentation on High Resolutional T2-Weighted MRI
# Estrada et al. Neuroimage 2021.
# https://doi.org/10.1016/j.neuroimage.2021.118464

951  L-olfactory-bulb                       0   0   255 0
952  R-olfactory-bulb                       255 0   0   0
```

m-reuter commented:
Then it will be easiest if we move to a different range, given that they have their labels in FreeSurfer now. I will discuss this with FreeSurfer.

dkuegler commented:
@santiestrada32 please also add your labels to this PR instead of #459 .

santiestrada32 commented:
@dkuegler Could you add this label range after the samseg-charm variable list

```
# Below is the color table for olfactory bulb structures generated by FastSurfer OB pipeline.
# See : https://doi.org/10.1016/j.neuroimage.2021.118464

941  L-olfactory-bulb                       0   0   255 0
942  R-olfactory-bulb                       255 0   0   0

# Below is the color table for the hypothalamic subregions generated by FastSurfer HypVINN pipeline
#See : https://doi.org/10.1162/imag_a_00034

951  R-N.opticus                            70  130 180 0
952  L-N.opticus                            130 180 70  0
953  R-C.mammilare                          205 62  78  0
954  R-Optic-tract                          80  120 134 0
955  L-Optic-tract                          196 58  250 0
956  L-C.mammilare                          0   148 0   0
957  R-Chiasma-Opticum                      220 248 164 0
958  L-Chiasma-Opticum                      230 148 34  0
959  Ant-Commisure                          10  180 225 0
960  3rd-Ventricle                          118 0   100 0
961  R-Fornix                               122 200 120 0
962  L-Fornix                               236 13  176 0
964  Epiphysis                              204 182 142 0
966  Hypophysis                             119 159 176 0
967  Infundibulum                           220 216 20  0
972  Tuberal-Region                         120 60  110 0
976  L-Med-Hypothalamus                     165 255 0   0
977  L-Lat-Hypothalamus                     0   255 127 0
978  L-Ant-Hypothalamus                     165 42  42  0
979  L-Post-Hypothalamus                    255 215 0   0
986  R-Med-Hypothalamus                     115 255 0   0
987  R-Lat-Hypothalamus                     60  255 127 0
988  R-Ant-Hypothalamus                     165 142 42  0
989  R-Post-Hypothalamus                    255 170 20  0
```

m-reuter commented:
why are OB labels moved? OBnet is released already, so these should be fixed?
Also why do you need such a large range for the labels? Please compress the IDS and maybe start at 960 ? 
But final numbers will be determined by my discussion with FreeSurfer anyway.

santiestrada32 commented:
Here is my new proposal
```
# Below is the color table for olfactory bulb structures generated by FastSurfer OB pipeline.
# See : https://doi.org/10.1016/j.neuroimage.2021.118464

951  L-olfactory-bulb                       0   0   255 0
952  R-olfactory-bulb                       255 0   0   0

# Below is the color table for the hypothalamic subregions generated by FastSurfer HypVINN pipeline
#See : https://doi.org/10.1162/imag_a_00034

961  R-N.opticus                            70  130 180 0
962  L-N.opticus                            130 180 70  0
963  R-C.mammilare                          205 62  78  0
964  R-Optic-tract                          80  120 134 0
965  L-Optic-tract                          196 58  250 0
966  L-C.mammilare                          0   148 0   0
967  R-Chiasma-Opticum                      220 248 164 0
968  L-Chiasma-Opticum                      230 148 34  0
969  Ant-Commisure                          10  180 225 0
970  3rd-Ventricle                          118 0   100 0
971  R-Fornix                               122 200 120 0
972  L-Fornix                               236 13  176 0
973  Epiphysis                              204 182 142 0
974  Hypophysis                             119 159 176 0
975  Infundibulum                           220 216 20  0
976  Tuberal-Region                         120 60  110 0
977  L-Med-Hypothalamus                     165 255 0   0
978  L-Lat-Hypothalamus                     0   255 127 0
979  L-Ant-Hypothalamus                     165 42  42  0
980  L-Post-Hypothalamus                    255 215 0   0
981  R-Med-Hypothalamus                     115 255 0   0
982  R-Lat-Hypothalamus                     60  255 127 0
983  R-Ant-Hypothalamus                     165 142 42  0
984  R-Post-Hypothalamus                    255 170 20  0

```

dkuegler committed:
Add OB and Hypothalamus labels

Add olfactory bulb labels from Estrada et al., Automated olfactory bulb segmentation on high resolutional T2-weighted MRI,
NeuroImage, 10.1016/j.neuroimage.2021.118464.

Add hypothalamus labels from Estrada et al,  FastSurfer-HypVINN: Automated sub-segmentation of the hypothalamus and adjacent structures on high-resolutional brain MRI, Imaging Neuroscience, 10.1162/imag_a_00034

dkuegler commented:
I added OB and Hypothalamus labels to the LUT.

m-reuter commented:
not necessary until final label ids are know. 
Anyway, we probably should change the name of this file to FastSurferColorLUT.txt as we may diverge (temporarily or permanently) from FreeSurfer and should not have two different files with the same name.

m-reuter commented:
@santiestrada32 can you check which label names might be duplicated. Probably 3rd-Ventricle , but also others?

dkuegler commented:
Question that should also be answered... Those labels that are duplicates, are the label definitions of those labels equivalent? Or are there major systematic differences.

santiestrada32 commented:
The only label that has the same name is the 3rd-Ventricle. The definition of the segmentation is different as our 3rd ventricle is more like a supporting label to the hypothalamus labels. To remove the problem of the naming, we can name our version as third-ventricle and not 3rd-Ventricle (Fs name).

I noticed that SAMSEG-CHARM + 400, also has the Optic-Nerve structure, so i don't know if the definition is the same as ours. Nevertheless, merging our LUT will not have a conflict as we have a different naming convention ( R-N.opticus, L-N.opticus)

This is the same case for the fornix, where the HiRes Hippocampus labeling method has the Right and left fornix, but the naming conventions are different

right_fornix --- R-Fornix (ours)
left_fornix   --- L-Fornix (ours)


So, my proposal for now will be to include our new variables with the following names.
```
# Below is the color table for olfactory bulb structures generated by FastSurfer OB pipeline.
# See : https://doi.org/10.1016/j.neuroimage.2021.118464

951  L-olfactory-bulb                       0   0   255 0
952  R-olfactory-bulb                       255 0   0   0

# Below is the color table for the hypothalamic subregions generated by FastSurfer HypVINN pipeline
#See : https://doi.org/10.1162/imag_a_00034

961  R-N.opticus                            70  130 180 0
962  L-N.opticus                            130 180 70  0
963  R-C.mammilare                          205 62  78  0
964  R-Optic-tract                          80  120 134 0
965  L-Optic-tract                          196 58  250 0
966  L-C.mammilare                          0   148 0   0
967  R-Chiasma-Opticum                      220 248 164 0
968  L-Chiasma-Opticum                      230 148 34  0
969  Ant-Commisure                          10  180 225 0
970  Third-Ventricle                          118 0   100 0
971  R-Fornix                               122 200 120 0
972  L-Fornix                               236 13  176 0
973  Epiphysis                              204 182 142 0
974  Hypophysis                             119 159 176 0
975  Infundibulum                           220 216 20  0
976  Tuberal-Region                         120 60  110 0
977  L-Med-Hypothalamus                     165 255 0   0
978  L-Lat-Hypothalamus                     0   255 127 0
979  L-Ant-Hypothalamus                     165 42  42  0
980  L-Post-Hypothalamus                    255 215 0   0
981  R-Med-Hypothalamus                     115 255 0   0
982  R-Lat-Hypothalamus                     60  255 127 0
983  R-Ant-Hypothalamus                     165 142 42  0
984  R-Post-Hypothalamus                    255 170 20  0

```

m-reuter commented on FastSurferCNN/config/FreeSurferColorLUT.txt:
```diff
@@ -470,6 +498,56 @@
+967  R-Chiasma-Opticum                      220 248 164 0
+968  L-Chiasma-Opticum                      230 148 34  0
+969  Ant-Commisure                          10  180 225 0
+970  3rd-Ventricle                          118 0   100 0
```
```suggestion
970  Third-Ventricle                        118 0   100 0
```

m-reuter committed:
Update FastSurferCNN/config/FreeSurferColorLUT.txt
